zkevm-circuits/src/evm_circuit/execution/error_return_data_oo_bound.rs

```diff
@@ -84,13 +83,23 @@ impl<F: Field> ExecutionGadget<F> for ErrorReturnDataOutOfBoundGadget<F> {
             return_data_length.expr(),
             from_bytes::expr(&end.cells[..N_BYTES_U64]),
         );
-        // Any of [offset_out_of_range, end_out_of_range, end_exceed_length] occurs.
-        cb.require_in_set(
-            "Any of [offset_out_of_range, end_out_of_range, end_exceed_length] occurs",
-            not::expr(is_data_offset_within_range.expr())
-                + not::expr(is_end_within_range.expr())
-                + is_end_exceed_length.expr(),
-            vec![1.expr(), 2.expr(), 3.expr()],
+
+        // `end > U256::MAX` is necessary to check with AddWordsGadget carry. Since
+        // offset plus length may exceed a Word and remainder (end) may be less
+        // than `u64::MAX` (e.g. Word::MAX + 1).
```

Comment:
How about comment change as below
```rust
// Need to assure `data + offset == reminder_end` in all situation
// once above satisfied then checking `reminder_end` is equivalent constraint `data + offset`. 
// To assure that, `AddWordsGadget carry` must no overflow.
```

Reply:
Updated this comment in [4ea268b](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1321/commits/4ea268b414ea757e452eab267e9a8a656c0cadcc).

@hero78119 may need help for review again. Thanks.